From b9db1ed4a901e9c0af7ac8cc5d4d933b5b9fd4b5 Mon Sep 17 00:00:00 2001
From: dje <dje@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Tue, 29 Nov 2016 17:50:27 +0000
Subject: [PATCH] PR libstdc++/68838 * testsuite/lib/libstdc++.exp
 (DEFAULT_CXXFLAGS): Add -Wl,-bmaxdata on AIX. *
 testsuite/23_containers/vector/profile/vector.cc: Remove
 dg-additional-options.

git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@242967 138bc75d-0d04-0410-961f-82ee72b054a4
---
 libstdc++-v3/ChangeLog                                        | 8 ++++++++
 libstdc++-v3/testsuite/23_containers/vector/profile/vector.cc | 2 --
 libstdc++-v3/testsuite/lib/libstdc++.exp                      | 3 +++
 3 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/libstdc++-v3/ChangeLog b/libstdc++-v3/ChangeLog
index 4a26191..02ad29e 100644
--- a/libstdc++-v3/ChangeLog
+++ b/libstdc++-v3/ChangeLog
@@ -1,3 +1,11 @@
+2016-11-29  David Edelsohn  <dje.gcc@gmail.com>
+
+	PR libstdc++/68838
+	* testsuite/lib/libstdc++.exp (DEFAULT_CXXFLAGS): Add -Wl,-bmaxdata on
+	AIX.
+	* testsuite/23_containers/vector/profile/vector.cc: Remove
+	dg-additional-options.
+
 2016-11-26  Tim Shen  <timshen@google.com>
 
 	PR libstdc++/78441
diff --git a/libstdc++-v3/testsuite/23_containers/vector/profile/vector.cc b/libstdc++-v3/testsuite/23_containers/vector/profile/vector.cc
index 9630a39..7806fb2 100644
--- a/libstdc++-v3/testsuite/23_containers/vector/profile/vector.cc
+++ b/libstdc++-v3/testsuite/23_containers/vector/profile/vector.cc
@@ -2,8 +2,6 @@
 // Advice: set tmp as 10000
 
 // { dg-options "-DITERATIONS=20" { target simulator } }
-// AIX requires higher memory limit
-// { dg-additional-options "-Wl,-bmaxdata:0x20000000" { target { powerpc-ibm-aix* } } }
 
 #ifndef ITERATIONS
 #define ITERATIONS 2000
diff --git a/libstdc++-v3/testsuite/lib/libstdc++.exp b/libstdc++-v3/testsuite/lib/libstdc++.exp
index bdb80a0..190f070 100644
--- a/libstdc++-v3/testsuite/lib/libstdc++.exp
+++ b/libstdc++-v3/testsuite/lib/libstdc++.exp
@@ -136,6 +136,9 @@ proc libstdc++_init { testfile } {
 	if { [string match "powerpc-*-darwin*" $target_triplet] } {
 	    append DEFAULT_CXXFLAGS " -multiply_defined suppress"
 	} 
+	if { [string match "powerpc-ibm-aix*" $target_triplet] } {
+	    append DEFAULT_CXXFLAGS " -Wl,-bmaxdata:0x20000000"
+	} 
     }
     v3track DEFAULT_CXXFLAGS 2
 
